ethcontract-common/src/bytecode.rs

```diff
@@ -17,20 +17,31 @@ use web3::types::{Address, Bytes};
 pub struct Bytecode(String);
 
 impl Bytecode {
-    /// Reads hex bytecode representation from a string slice.
+    /// Reads hex bytecode representation from a string or map slice.
     pub fn from_hex_str(s: &str) -> Result<Self, BytecodeError> {
         if s.is_empty() {
             // special case where we have an empty string byte code.
             return Ok(Bytecode::default());
         }
 
+        let s = if s.starts_with('{') {
```

Comment:
This seems wrong to me. Specifically the Foundry artifact output uses a map and not a string for bytecode (in fact, the actual bytecode in the map has the same format that this type expects).

Maybe the solution is to change the `Hardhat` artifact type to accept `ByteCode | { object: Bytecode }` for its bytecode field (and not change the `Bytecode` parsing logic itself)?